zeppelin: web application FAILURE on Spark 2.1.

Zeppelin now supports Spark 2.1.

using ZEPPELIN_COMMIT 019df1f6bc4875bd8592d2cb77164b1c9756ac1c

Error SEEN:
**************
[INFO] [3/4] Linking dependencies...
[WARNING] warning Unmet peer dependency "eslint@>=3.8.1".
[INFO] [4/4] Building fresh packages...
[INFO] $ bower install --silent
[ERROR] [{
[ERROR]   "code": "ESUDO",
[ERROR]   "details": "Since bower is a user command, there is no need to execute it with superuser permissions.\nIf you're having permission errors when using bower without sudo, please spend a few minutes learning more about how your system should work and make any necessary repairs.\n\nhttp://www.joyent.com/blog/installing-node-and-npm\nhttps://gist.github.com/isaacs/579814\n\nYou can however run a command with sudo using --allow-root option",
[ERROR]   "id": "ESUDO",
[ERROR]   "level": "error",
[ERROR]   "data": {},
[ERROR]   "message": "Cannot be run with sudo",
[ERROR]   "stacktrace": "Error: Cannot be run with sudo\n    at createError (/usr/src/zeppelin/zeppelin-web/node_modules/bower/lib/util/createError.js:4:15)\n    at rootCheck (/usr/src/zeppelin/zeppelin-web/node_modules/bower/lib/util/rootCheck.js:26:24)\n    at Object.<anonymous> (/usr/src/zeppelin/zeppelin-web/node_modules/bower/lib/bin/bower.js:33:1)\n    at Module._compile (module.js:570:32)\n    at Object.Module._extensions..js (module.js:579:10)\n    at Module.load (module.js:487:32)\n    at tryModuleLoad (module.js:446:12)\n    at Function.Module._load (module.js:438:3)\n    at Module.require (module.js:497:17)\n    at require (internal/module.js:20:19)"
[ERROR] }]
[ERROR] error Command failed with exit code 1.
[INFO] info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Zeppelin: web Application .......................... FAILURE [01:48 min]
[INFO] Zeppelin: Server ................................... SKIPPED
[INFO] Zeppelin: Packaging distribution ................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:50 min
[INFO] Finished at: 2017-02-02T00:01:33+00:00
[INFO] Final Memory: 27M/208M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.3:yarn (yarn install) on project zeppelin-web: Failed to run task: 'yarn install --no-lockfile' failed. (error code 1) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.3:yarn (yarn install) on project zeppelin-web: Failed to run task
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoFailureException: Failed to run task
	at com.github.eirslett.maven.plugins.frontend.mojo.AbstractFrontendMojo.execute(AbstractFrontendMojo.java:95)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: com.github.eirslett.maven.plugins.frontend.lib.TaskRunnerException: 'yarn install --no-lockfile' failed. (error code 1)
	at com.github.eirslett.maven.plugins.frontend.lib.YarnTaskExecutor.execute(YarnTaskExecutor.java:61)
	at com.github.eirslett.maven.plugins.frontend.mojo.YarnMojo.execute(YarnMojo.java:65)
	at com.github.eirslett.maven.plugins.frontend.mojo.AbstractFrontendMojo.execute(AbstractFrontendMojo.java:89)
	... 22 more

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM gettyimages/spark:2.0.2-hadoop-2.7
+FROM gettyimages/spark:2.1.0-hadoop-2.7
 
 # SciPy
 RUN set -ex \
@@ -29,11 +29,13 @@ ENV ZEPPELIN_PORT 8080
 ENV ZEPPELIN_HOME /usr/zeppelin
 ENV ZEPPELIN_CONF_DIR $ZEPPELIN_HOME/conf
 ENV ZEPPELIN_NOTEBOOK_DIR $ZEPPELIN_HOME/notebook
-ENV ZEPPELIN_COMMIT 7f6f739ae396e07de573bea4ef16a388c54e77b8
+ENV ZEPPELIN_COMMIT v0.7.0
+RUN echo '{ "allow_root": true }' > /root/.bowerrc
 RUN set -ex \
  && buildDeps=' \
     git \
     bzip2 \
+    npm \
  ' \
  && apt-get update && apt-get install -y --no-install-recommends $buildDeps \
  && curl -sL http://archive.apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.tar.gz \

--- a/README.md
+++ b/README.md
@@ -4,9 +4,9 @@ A `debian:jessie` based Spark and [Zeppelin](http://zeppelin.apache.org) Docker 
 
 This image is large and opinionated. It contains:
 
-- [Spark 2.0.2](http://spark.apache.org/docs/2.0.2) and [Hadoop 2.7.3](http://hadoop.apache.org/docs/r2.7.3)
+- [Spark 2.1.0](http://spark.apache.org/docs/2.1.0) and [Hadoop 2.7.3](http://hadoop.apache.org/docs/r2.7.3)
 - [PySpark](http://spark.apache.org/docs/2.0.1/api/python) support with [Python 3.4](https://docs.python.org/3.4), [NumPy](http://www.numpy.org), [PandaSQL](https://github.com/yhat/pandasql), and [SciPy](https://www.scipy.org/scipylib/index.html), but no matplotlib.
-- A partial list of interpreters out-of-the-box. If your favorite interpreter isn't included, consider [adding it with the api](http://zeppelin.apache.org/docs/0.7.0-SNAPSHOT/manual/dynamicinterpreterload.html).
+- A partial list of interpreters out-of-the-box. If your favorite interpreter isn't included, consider [adding it with the api](http://zeppelin.apache.org/docs/0.7.0/manual/dynamicinterpreterload.html).
   - spark
   - shell
   - angular

